contxt/services/ems.py

```diff
@@ -1,102 +1,100 @@
-from datetime import datetime
-from typing import Optional
+from datetime import date, timedelta
+from typing import List, Optional
 
-from contxt.legacy.services import GET, APIObjectCollection, Service
+from contxt.auth import Auth
 from contxt.models.ems import (
-    FacilityMainService,
-    FacilityUtilitySpend,
-    FacilityUtilityUsage,
+    Facility,
+    MainService,
+    ResourceType,
+    UtilitySpend,
+    UtilityUsage,
 )
+from contxt.services.api import ApiEnvironment, ConfiguredApi
 
-CONFIGS_BY_ENVIRONMENT = {
-    "production": {
-        "base_url": "https://ems.api.ndustrial.io/",
-        "audience": "e2IT0Zm9RgGlDBkLa2ruEcN9Iop6dJAS",
-    },
-    "staging": {
-        "base_url": "https://ems-staging.api.ndustrial.io/",
-        "audience": "vMV67yaRFgjBB1JFbT3vXBOlohFdG1I4",
-    },
-}
 
-
-class EMSService(Service):
+class EmsService(ConfiguredApi):
     """
     Service to interact with our EMS API.
     """
 
-    def __init__(self, auth, environment="production"):
-
-        if environment not in CONFIGS_BY_ENVIRONMENT:
-            raise Exception("Invalid environment specified")
-
-        self.env = CONFIGS_BY_ENVIRONMENT[environment]
-
-        super().__init__(
-            base_url=self.env["base_url"],
-            access_token=auth.get_token_provider(self.env["audience"]).access_token,
-        )
-
-    def get_main_services(self, facility_id: int, type: Optional[str] = None):
-
-        response = self.execute(GET(uri=f"facilities/{facility_id}"), execute=True)
-
-        # manual filter for main services
-        rows = []
-        if type:
-            for row in response["main_services"]:
-                if row["type"] == type:
-                    rows.append(row)
-        else:
-            rows = response["main_services"]
-
-        return APIObjectCollection([FacilityMainService(row) for row in rows])
+    _envs = (
+        ApiEnvironment(
+            name="production",
+            base_url="https://ems.api.ndustrial.io/v1",
+            client_id="e2IT0Zm9RgGlDBkLa2ruEcN9Iop6dJAS",
+        ),
+        ApiEnvironment(
+            name="staging",
+            base_url="https://ems-staging.api.ndustrial.io/v1",
+            client_id="vMV67yaRFgjBB1JFbT3vXBOlohFdG1I4",
+        ),
+    )
+
+    def __init__(self, auth: Auth, env: str = "production"):
+        super().__init__(env, auth)
+
+    def get_facility(self, id: int) -> Facility:
+        return Facility.from_api(self.get(f"facilities/{id}"))
+
+    def get_main_services(
+        self, facility_id: int, resource_type: Optional[ResourceType] = None
+    ) -> List[MainService]:
+        facility = self.get_facility(facility_id)
+        main_services = facility.main_services
+
+        # Manually filter on resource type
+        if resource_type:
+            main_services = [
```

Comment:
did flake/black really format the below like that?

Reply:
yep